wear/src/main/java/com/example/k_dual/watchface/KDualCanvasRender.kt

```diff
@@ -211,30 +284,79 @@ class KDualCanvasRender (
             arrowSrc = arrowUp
         }
 
-        // TODO. png resolution issue
+        val arrowPaint = if (isAlertOn) {
+            CustomPaint.arrowPaint(color)
+        } else { null }
+
         val boxPaddingHorizontal = unitF * 8f
         val rectCenterY = rectTop + rectHeight/2
         val arrowSize = unitF * 15f
         val arrowRect = RectF(rectLeft + boxPaddingHorizontal, rectCenterY - arrowSize/2, rectLeft + boxPaddingHorizontal + arrowSize, rectCenterY + arrowSize/2)
-        canvas.drawBitmap(arrowSrc, null, arrowRect, null)
+        canvas.drawBitmap(arrowSrc, null, arrowRect, arrowPaint)
 
         // Draw difference value
-        val paint = CustomPaint.differenceTextPaint(unitF, robotoRegular)
+        val paint = if (isAlertOn) { CustomPaint.differenceTextPaint(unitF, robotoRegular, color) } else {
+            CustomPaint.differenceTextPaint(unitF, robotoRegular, null)
+        }
         paint.getTextBounds(differenceText, 0, differenceText.length, textBounds)
         canvas.drawText(differenceText, rectLeft+rectWidth-boxPaddingHorizontal-textBounds.width(), rectCenterY-textBounds.exactCenterY(), paint)
     }
 
-    private fun drawBloodGlucose(canvas: Canvas, order: Number, value: Int) {
-        val paint = CustomPaint.bloodGlucoseTextPaint(unitF, robotoMedium)
+    private fun drawBloodGlucose(canvas: Canvas, order: Number?, value: Int) {
+        val paint = CustomPaint.bloodGlucoseTextPaint(unitF, isDualMode, robotoMedium)
         val valueText = value.toString()
         paint.getTextBounds(valueText, 0, valueText.length, textBounds)
 
-        val rectHalfHeight = unitF * 27f / 2f
-        val textX = width - mHorizontal - pHorizontal - unitF*69f/2
-        val textY = if (order == 1) { height/2 - gap - pBottom - rectHalfHeight + textBounds.height()/2 } else { height/2 + gap + pBottom + textBounds.height() - rectHalfHeight + textBounds.height()/2 }
+        val textX: Float
+        val textY: Float
+
+        if (isDualMode) {
+            val rectHalfHeight = unitF * 27f / 2f
+            textX = width - mHorizontal - pHorizontal - unitF*69f/2
+            textY = if (order == 1) {
+                        height/2 - gap - pBottom - rectHalfHeight + textBounds.height()/2
+                    } else {
+                        height/2 + gap + pBottom + textBounds.height() - rectHalfHeight + textBounds.height()/2
+                    }
+        } else {
+            textX = width/2
+            textY = height/2 + textBounds.height()/2
+        }
+
         canvas.drawText(valueText, textX, textY, paint)
     }
 
+    override fun onTapEvent(tapType: Int, tapEvent: TapEvent, complicationSlot: ComplicationSlot?) {
+        // For test blink effect
+        if (tapType == TapType.DOWN) {
+            blinkEffect(1)
+        }
+    }
+
+    // Function for low/high alert highlights
+    private fun blinkEffect(order: Number?) {
```

Comment:
[ChatGPT] **Kotlin Coroutines for Blink Effect**: The blinkEffect function uses a Handler for post-delayed calls. Using Kotlin Coroutines, you can simplify this code and make it more readable:
```kotlin
private fun blinkEffect(order: Number?) = CoroutineScope(Dispatchers.Main).launch {
    repeat(3) {
        delay(if (it % 2 == 0) 400L else 100L)
        when (order) {
            2 -> isUser2AlertOn = !isUser2AlertOn
            else -> isUser1AlertOn = !isUser1AlertOn
        }
    }
    isUser1AlertOn = false
    isUser2AlertOn = false
}
```